spec/datadog/tracing/metadata/metastruct_spec.rb

```diff
@@ -0,0 +1,64 @@
+require 'datadog/tracing/metadata/metastruct'
+
+RSpec.describe Datadog::Tracing::Metadata::Metastruct do
+  subject(:metastruct) { test_object.instance_variable_get(:@metastruct) }
+  let(:test_object) { described_class.new(preexisting_metastruct) }
+  let(:preexisting_metastruct) { {} }
+
+  describe '.empty' do
+    let(:test_object) { described_class.empty }
+
+    it { is_expected.to eq({}) }
+  end
+
+  describe '#initialize' do
+    context 'when setting meta struct' do
+      context 'with empty metastruct' do
+        it { is_expected.to eq({}) }
+      end
+
+      context 'with not empty metastruct' do
+        let(:preexisting_metastruct) { { 'key' => 'value' } }
+
+        it { is_expected.to eq({ 'key' => 'value' }) }
+      end
+    end
+  end
+
+  describe '#deep_merge!' do
```

Comment:
I think we should also test deep merging - now we only test merging of flat hashes